sqlglot/dialects/clickhouse.py

```diff
@@ -776,6 +777,14 @@ def _parse_expression(self) -> t.Optional[exp.Expression]:
 
             return this
 
+        def _parse_columns(self) -> exp.Expression:
+            this: exp.Expression = self.expression(exp.Columns, this=self._parse_lambda())
+
+            while self._next and self._match_text_seq(")", "APPLY", "("):
+                self._match(TokenType.R_PAREN)
+                this = exp.Apply(this=this, expression=self._parse_var(any_token=True))
```

Comment:
Can this only be applied to `COLUMNS`? I was thinking we could add this after `_parse_column` or something, to make it more generic and not have to deal with special cases again. I'd also try to avoid duplicating this logic and/or consolidate the `COLUMNS` parsing with what we did for duckdb if possible, looks like there may be some overlap.

Reply:
From my understanding yes, the `APPLY` seems to follow `*` and `COLUMNS`, it doesn't work with any expression thus the specialized solution. However, what I didn't expect in the initial PR is that `COLUMNS(...)` can be used in other scenarios as well.

Note that the recent DuckDB PR added support `SELECT *COLUMNS(...)` which is a special case/operator (column unpacking), not sure if these should be consolidated in one representation.

Reply:
What's `COLUMNS` used for in CH? It looks like it also does dynamic column expansion, hence my suggestion. I'd try out DuckDB's syntax `*COLUMNS(...)` to see if it works for CH as well, but even if it doesn't I'd still consider consolidating them if they represent the same thing.

Reply:
DuckDB also supports `COLUMNS(...)` [[1]](https://duckdb.org/docs/sql/query_syntax/select.html#star-expressions), the unpacking operator is so that the expansion is done into a function [[2]](https://duckdb.org/2024/09/09/announcing-duckdb-110.html#unpacked-columns).

Reply:
@georgesittas it is a dynamic column selection https://clickhouse.com/blog/sql-dynamic-column-selection-clickhouse

Reply:
Also consequent applys are also valid like this 

```
  COLUMNS('.*_amount|fee|tax')
  APPLY(avg)
  APPLY(round)
```

Reply:
> DuckDB also supports `COLUMNS(...)` [[1]](https://duckdb.org/docs/sql/query_syntax/select.html#star-expressions), the unpacking operator is so that the expansion is done into a function [[2]](https://duckdb.org/2024/09/09/announcing-duckdb-110.html#unpacked-columns).

I see, makes sense I guess. Ok cool, so can we still consolidate the existing `APPLY (...)` implementations though?

Reply:
Not sure, because e.g. for the projection list we parse expressions (i.e star or `COLUMNS`) but if `COLUMNS(..)` is picked up by `_parse_function()`, the subsequent `APPLY(...)` won't be picked. 

We could try to move `apply` to `parse_star_ops()` so that it's at least removed from `_parse_expressions()`.